### PR TITLE
IO-719: Fixed error of copying directories between different file systems

### PR DESCRIPTION
**Pull request by ashcheglov**
Also fixed relativize error for different types of paths
(absolute/relative)

### CODE REVIEW

**Review by garydgregory — CHANGES_REQUESTED** [review]
No tabs please. See broken build.

**Comment by ashcheglov** [comment]
Looks like all checks passed, excluding early access environments

File: src/main/java/org/apache/commons/io/file/CopyDirectoryVisitor.java [hunk 1]
```diff
@@ -153,7 +155,7 @@ public FileVisitResult preVisitDirectory(final Path directory, final BasicFileAt
 
     @Override
     public FileVisitResult visitFile(final Path sourceFile, final BasicFileAttributes attributes) throws IOException {
-        final Path targetFile = targetDirectory.resolve(sourceDirectory.relativize(sourceFile));
+        final Path targetFile = targetDirectory.resolve(sourceDirectory.relativize(sourceFile).toString());
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/io/file/CopyDirectoryVisitor.java:158]: As above...

**Comment by garydgregory** [comment]
@ashcheglov 
The jar file you refer to in the new tests does not exist, you should put in in the same place as the other test fixtures: `/src/test/resources/org/apache/commons/io`.

**Comment by garydgregory** [comment]
@ashcheglov 
Thank  you for your PR, I applied the patch in two separate commits with additional clean ups and minor refactoring. Please see git master and close this PR if your use case is satified.

**Comment by ashcheglov** [comment]
Cool, thanks!

**Pull request closed without merging**